packages/bundler/src/Bundler.ts

```diff
@@ -42,19 +66,61 @@ export class Bundler implements IBundler {
      * @returns Promise<UserOpResponse>
      */
     async sendUserOp(userOp: UserOperation, chainId: ChainId): Promise<UserOpResponse> {
-        const hexifiedUserOp = deepHexlify(await resolveProperties(userOp))        
-        let params = [hexifiedUserOp, this.bundlerConfig.epAddress, chainId]
-        const response: UserOpResponse = await sendRequest({
-            url: `${this.bundlerConfig.bundlerUrl}`,
+        // transformUserOP will convert all bigNumber values to string
+        userOp = transformUserOP(userOp)
+        const hexifiedUserOp = deepHexlify(await resolveProperties(userOp))
+        let params = [hexifiedUserOp, this.bundlerConfig.epAddress]
+        const sendUserOperationResponse: SendUserOperationResponse = await sendRequest({
+            url: `${this.bundlerConfig.bundlerUrl}/${chainId}/${this.bundlerConfig.apiKey}`,
```

Comment:
URL generation is repeated three times in this file in same way.

There should be an internal function that just returns the final url for a given chainId. Then we call this function from each method.